editor/editor_file_system.cpp

```diff
@@ -2773,6 +2773,34 @@ void EditorFileSystem::reimport_file_with_custom_parameters(const String &p_file
 	emit_signal(SNAME("resources_reimported"), reloads);
 }
 
+void EditorFileSystem::_queue_refresh_filesystem() {
+	if (refresh_queued) {
+		return;
+	}
+	refresh_queued = true;
+	get_tree()->connect(SNAME("process_frame"), callable_mp(this, &EditorFileSystem::_refresh_filesystem), CONNECT_ONE_SHOT);
+}
+
+void EditorFileSystem::_refresh_filesystem() {
+	for (EditorFileSystemDirectory *dir : folders_to_refresh) {
+		dir->subdirs.sort_custom<DirectoryComparator>();
+	}
+	folders_to_refresh.clear();
+
+	if (!files_to_refresh.is_empty()) {
+		update_files(files_to_refresh);
+		files_to_refresh.clear();
+	}
+
+	if (!reimport_files_to_refresh.is_empty()) {
+		reimport_files(reimport_files_to_refresh);
+		reimport_files_to_refresh.clear();
+	}
+
+	emit_signal(SNAME("filesystem_changed"));
```

Comment:
The `filesystem_changed` signal could be emitted twice, once in `update_files` and once here.